internal/lexer/lexer_test.go

```diff
@@ -0,0 +1,83 @@
+package lexer_test
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+	"text/scanner"
+
+	"github.com/neelance/graphql-go/internal/lexer"
+)
+
+func TestLexer(t *testing.T) {
```

Comment:
This test looks overly complicated. I also think that we should test the public API of the `graphql` package instead of this internal API. Could you come up with some failing test there that gets fixed by your change to `ConsumeFloat`?